src/gwt/src/org/rstudio/studio/client/workbench/views/environment/EnvironmentPane.java

```diff
@@ -415,8 +415,17 @@ private void executeFunctionForObject(String function, String objectName)
    private Widget createImportMenu()
    {
       ToolbarPopupMenu menu = new ToolbarPopupMenu();
-      menu.addItem(commands_.importDatasetFromFile().createMenuItem(false));
-      menu.addItem(commands_.importDatasetFromURL().createMenuItem(false));
+      
+      if (prefs_.useDataImport().getValue())
+      {
+         menu.addItem(commands_.importDatasetFromCSV().createMenuItem(false));
```

Comment:
Once you have the remove logic in place then you can actually add all of these commands to the menu unqualified (the removed ones won't show up). You'll also want to add your new commands to the main menu here: https://github.com/rstudio/rstudio/blob/master/src/gwt/src/org/rstudio/studio/client/workbench/commands/Commands.cmd.xml#L363-L366. Again, we can specify all the old and new items there then the removed ones will wash out.